src/window_handle.rs

```diff
@@ -624,22 +625,22 @@ impl WindowHandle {
             }
 
             if self.needs_style() {
-                flags |= ChangeFlags::STYLE;
+                paint = true;
                 self.style();
             }
 
             if self.needs_layout() {
-                flags |= ChangeFlags::LAYOUT;
+                paint = true;
                 self.layout();
             }
 
-            flags |= self.process_deferred_update_messages();
-            flags |= self.process_anim_update_messages();
+            self.process_deferred_update_messages();
+            self.process_anim_update_messages();
         }
 
         self.set_cursor();
 
-        if !flags.is_empty() {
+        if paint {
```

Comment:
What if a view just wanted to request_paint without style or layout change, e.g. a canvas?

Reply:
They can just call `request_paint` to do that.

Reply:
Ah I see it ```request_style``` now. But is it efficient enough when the canvas wants some animation frames.

Reply:
It only requests style on one view, but I'm going to replace it to just requesting painting.